### DISCUSSION TITLE
Garbage collection

### DISCUSSION THREAD

**Discussion by peter7891** [traefik/yaegi#1550] [Q&A] [answered]
Does this thing have a GC?
Since you claim the Go spec is implemented, it needs escape analysis and that leads to having a GC.
Can you point me to that code?

**Comment by mvertes** [comment, marked as answer]
Yaegi doesn’t implement escape analysis nor has its own GC. Instead it uses the GC from the Go runtime, because it runs on top of it and all the objects used and generated by the interpreter (through `reflect`) are already handled. We would need it, of course, if we had to target something else than the Go runtime.

As far as I know, the Go spec doesn't mandate a garbage collector.
